src/System.Runtime/tests/System/DecimalTests.cs

```diff
@@ -1621,9 +1637,26 @@ static decimal[] GetRandomData(out BigDecimal[] bigDecimals)
                             if (!unique.Add(d.ToString(CultureInfo.InvariantCulture)))
                                 continue; // skip duplicates
                             list.Add(d);
+
+                            if (hash)
+                            {
+                                // generate all possible variants of the number up-to max decimal scale
+                                for (var lastScale = scale; lastScale < 28;)
+                                {
+                                    d *= 1.0m;
+                                    unsafe
+                                    {
+                                        var curScale = (byte)(*(uint*)&d >> BigDecimal.ScaleShift);
+                                        if (curScale <= lastScale)
+                                            break;
+                                        lastScale = curScale;
+                                    }
+                                    list.Add(d);
+                                }
```

Comment:
Does this significantly impact the running time of the tests?

Reply:
No, it takes 6ms to run the GetHashCode tests and other tests aren't affected.

Reply:
Cool, thanks.